controllers/glance_controller.go

```diff
@@ -156,16 +156,18 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (res
 		instance.Status.Conditions = condition.Conditions{}
 	}
 
-	// initialize conditions used later as Status=Unknown
+	// initialize conditions used later as Status=Unknown, except the ReadyCondition
+	// that should be False when we start
 	cl := condition.CreateList(
+		// Mark ReadyCondition as False from the beginning
+		condition.FalseCondition(condition.ReadyCondition, condition.InitReason, condition.SeverityInfo, condition.ReadyInitMessage),
```

Comment:
as the reconcile loop always exit via the logic in the deferred call https://github.com/openstack-k8s-operators/glance-operator/blob/ca211863b3824313ae1794af92179b8a4a44214f/controllers/glance_controller.go#L129-L139 this should not be necessary as the deferred call will either set the ReadyCondition to True all mirror the status of the most sever condition in the list.

Reply:
right, I'm just not there yet because I'm not sure that re-asserting all the conditions is the right thing to do at the end of the reconcile loop. I'm going to update this patch soon, and the plan is to remove this evaluation from the defer call

Reply:
There is a side conversation on the meaning of the `isReady` here, that's why I did not update this code so far.